Setting Geocoordinate Highlight Returns a 500 error

Any attempt to set a geocoordinate highlight by any means available (via geocoordinate facet selection, via lex bar or directly against the api POST) returns a 500 whose errors imply that the latitude_longitude column of that facet doesn't exist despite it being a valid variable that displays on the front end and returns from the api. 

Example Payload:
`{
   "highlights":[{
       "key":"longitude_latitude",
       "type":"geobounds",
       "mode":"include",
       "minX":32,
       "maxX":42,
       "minY":33, 
       "maxY":39,
       "displayName":"Geocoordinate"
  }],
  "variables":["longitude_latitude","Country_ISO_Number"],
  "filters":[],
  "size":0,
  "dataMode":null
}`

As a post against:
`http://0.0.0.0:8080/distil/variable-summary/acled_clean/Notes/false/default`

Front End Error Information:
`xhr.js:177 POST http://0.0.0.0:8080/distil/variable-summary/acled_clean/Notes/true/default 500 (Internal Server Error)
actions.ts:1038 Error: Request failed with status code 500
    at createError (createError.js:16)
    at settle (settle.js:17)
    at XMLHttpRequest.handleLoad (xhr.js:62)`
    
Back End Error Information:
`[ Mar  2 12:02:16 ] [ WARN ] POST /distil/variable-summary/acled_clean/Notes/true/default 500 in 297.75ms (api/middleware/request_logger.go:130)
[ Mar  2 12:02:17 ] [ ERROR ] ERROR: column "longitude_latitude" does not exist (SQLSTATE 42703)
failed to fetch text histogram for variable summaries from postgres)`

--- a/api/model/grouped_variables.go
+++ b/api/model/grouped_variables.go
@@ -88,6 +88,9 @@ func ExpandFilterParams(dataset string, filterParams *FilterParams, includeHidde
 				} else if model.IsGeoBounds(variable.Type) {
 					gbg := variable.Grouping.(*model.GeoBoundsGrouping)
 					componentVars = append(componentVars, gbg.CoordinatesCol)
+				} else if model.IsBivariate(variable.Type) {
+					gcg := variable.Grouping.(*model.GeoCoordinateGrouping)
+					componentVars = append(componentVars, gcg.XCol, gcg.YCol)
 				}
 
 				// include the grouping ID if present

--- a/api/routes/grouping.go
+++ b/api/routes/grouping.go
@@ -237,7 +237,7 @@ func createGrouping(dataset string, storageName string, groupingType string, raw
 
 		// No key required in this case.
 		groupingVarName := strings.Join([]string{gcg.XCol, gcg.YCol}, task.DefaultSeparator)
-		err = meta.AddGroupedVariable(dataset, groupingVarName, "Geocoordinate", model.GeoCoordinateType, model.VarDistilRoleGrouping, gcg)
+		err = meta.AddGroupedVariable(dataset, groupingVarName, "Geocoordinate", model.BivariateFilter, model.VarDistilRoleGrouping, gcg)
 		if err != nil {
 			return err
 		}

--- a/go.mod
+++ b/go.mod
@@ -33,7 +33,7 @@ require (
 	github.com/russross/blackfriday v2.0.0+incompatible
 	github.com/shurcooL/sanitized_anchor_name v1.0.0 // indirect
 	github.com/stretchr/testify v1.6.1
-	github.com/uncharted-distil/distil-compute v0.0.0-20210302220240-fdbbb9f124a7
+	github.com/uncharted-distil/distil-compute v0.0.0-20210305172615-0aa0bbc089c8
 	github.com/uncharted-distil/distil-image-upscale v0.0.0-20210223142454-e2adf2b3dbb6
 	github.com/uncharted-distil/gdal v0.0.0-20200504224203-25f2e6a0dc2a
 	github.com/unchartedsoftware/plog v0.0.0-20200807135627-83d59e50ced5

--- a/go.mod
+++ b/go.mod
@@ -33,7 +33,7 @@ require (
 	github.com/russross/blackfriday v2.0.0+incompatible
 	github.com/shurcooL/sanitized_anchor_name v1.0.0 // indirect
 	github.com/stretchr/testify v1.6.1
-	github.com/uncharted-distil/distil-compute v0.0.0-20210305172615-0aa0bbc089c8
+	github.com/uncharted-distil/distil-compute v0.0.0-20210305185005-465699a38ed3
 	github.com/uncharted-distil/distil-image-upscale v0.0.0-20210223142454-e2adf2b3dbb6
 	github.com/uncharted-distil/gdal v0.0.0-20200504224203-25f2e6a0dc2a
 	github.com/unchartedsoftware/plog v0.0.0-20200807135627-83d59e50ced5

--- a/public/components/TypeChangeMenu.vue
+++ b/public/components/TypeChangeMenu.vue
@@ -84,6 +84,7 @@ import {
   EXPAND_ACTION_TYPE,
   EXPLODE_ACTION_TYPE,
   isTimeSeriesType,
+  isGeoLocatedType,
 } from "../util/types";
 import { hasFilterInRoute } from "../util/filters";
 import { createRouteEntry } from "../util/routes";
@@ -280,9 +281,12 @@ export default Vue.extend({
         // For timeseries, exploding one variable explodes them all
         const toRemove = datasetGetters
           .getGroupings(this.$store)
-          .filter(
-            (g) => isTimeSeriesType(g.colType) && g.datasetName === this.dataset
-          );
+          .filter((g) => {
+            return (
+              (isTimeSeriesType(g.colType) || isGeoLocatedType(g.colType)) &&
+              g.datasetName === this.dataset
+            );
+          });
 
         for (const g of toRemove) {
           // CDB: This needs to be converted into an API call that can handle removal of

--- a/public/components/facets/GeocoordinateFacet.vue
+++ b/public/components/facets/GeocoordinateFacet.vue
@@ -75,7 +75,6 @@ import {
 } from "../../store/dataset/module";
 import { getters as routeGetters } from "../../store/route/module";
 import { actions as appActions } from "../../store/app/module";
-import { Dictionary } from "../../util/dict";
 import {
   TableRow,
   VariableSummary,
@@ -112,7 +111,6 @@ const SINGLE_FIELD = 1;
 const SPLIT_FIELD = 2;
 const CLOSE_BUTTON_CLASS = "geo-close-button";
 const CLOSE_ICON_CLASS = "fa-times";
-const LON_LAT_KEY = "longitude:latitude";
 
 interface GeoField {
   type: number;
@@ -130,8 +128,6 @@ interface BucketData {
   extrema: Extrema;
   buckets: Bucket[];
 }
-
-const GEOCOORDINATE_LABEL = "longitude";
 
 /**
  * Geocoordinate Facet.

--- a/public/util/highlights.ts
+++ b/public/util/highlights.ts
@@ -8,6 +8,7 @@ import {
   VECTOR_FILTER,
   INCLUDE_FILTER,
   TEXT_FILTER,
+  BIVARIATE_FILTER,
 } from "../util/filters";
 import { getters as routeGetters } from "../store/route/module";
 import { getters as datasetGetters } from "../store/dataset/module";
@@ -164,7 +165,7 @@ export function createFiltersFromHighlights(
     ) {
       return {
         key: key,
-        type: GEOBOUNDS_FILTER,
+        type: variable.colType,
         mode: mode,
         minX: highlight.value.minX,
         maxX: highlight.value.maxX,

--- a/public/util/lex.ts
+++ b/public/util/lex.ts
@@ -25,6 +25,7 @@ import {
   TEXT_FILTER,
   GEOBOUNDS_FILTER,
   GEOCOORDINATE_FILTER,
+  BIVARIATE_FILTER,
 } from "./filters";
 import {
   LabelState,
@@ -170,7 +171,7 @@ export function filterParamsToLexQuery(
   const suggestions = variablesToLexSuggestions(activeVariables);
 
   const lexQuery = lexableElements.map((f, i) => {
-    if (f.type === GEOBOUNDS_FILTER) {
+    if (f.type === GEOBOUNDS_FILTER || f.type === BIVARIATE_FILTER) {
       return {
         field: suggestions[i],
         minX: new ValueStateValue(f.minX),
@@ -257,7 +258,11 @@ export function lexQueryToFiltersAndHighlight(
         value: {},
       } as Highlight;
 
-      if (type === GEOBOUNDS_FILTER || type === GEOCOORDINATE_FILTER) {
+      if (
+        type === GEOBOUNDS_FILTER ||
+        type === GEOCOORDINATE_FILTER ||
+        type === BIVARIATE_FILTER
+      ) {
         highlight.key = highlight.key;
         highlight.value.minX = parseFloat(lq.minX.key);
         highlight.value.maxX = parseFloat(lq.maxX.key);

--- a/public/util/types.ts
+++ b/public/util/types.ts
@@ -4,6 +4,7 @@ import { D3M_INDEX_FIELD } from "../store/dataset/index";
 import { getters as datasetGetters } from "../store/dataset/module";
 import store from "../store/store";
 import { Dictionary } from "./dict";
+import { BIVARIATE_FILTER } from "./filters";
 
 export enum DISTIL_ROLES {
   Grouping = "grouping",
@@ -86,6 +87,7 @@ export const META_TYPES = {
 };
 
 const TYPES_TO_LABELS: Dictionary<string> = {
+  [BIVARIATE_FILTER]: "Bivariate",
   [INTEGER_TYPE]: "Integer",
   [REAL_TYPE]: "Decimal",
   [REAL_LIST_TYPE]: "List",
@@ -198,7 +200,7 @@ const TEXT_SIMPLE_TYPES = [
   BOOL_TYPE,
 ];
 
-const GEOLOCATED_TYPES = [GEOCOORDINATE_TYPE, GEOBOUNDS_TYPE];
+const GEOLOCATED_TYPES = [GEOCOORDINATE_TYPE, GEOBOUNDS_TYPE, BIVARIATE_FILTER];
 
 const IMAGE_TYPES = [IMAGE_TYPE, MULTIBAND_IMAGE_TYPE];
 

